# Tugas Besar - IF2220 - Probabilitas dan Statistika

## Soal

Diberikan sebuah data Gandum.csv yang dapat diakses pada utas berikut: Dataset Tugas Besar IF2220. Gandum.csv merupakan data klasifikasi biji gandum yang mengandung 12 kolom sebagai berikut:
1. id
2. Daerah
3. SumbuUtama
4. SumbuKecil
5. Keunikan
6. AreaBulatan
7. Diameter
8. KadarAir
9. Keliling
10. Bulatan
11. Ransum
12. Kelas

Kolom 2-11 adalah kolom atribut (non-target), sedangkan kolom 12 adalah kolom target.
Anda diminta untuk melakukan analisis statistika sebagai berikut:

### 1.Menulis deskripsi statistika (Descriptive Statistics) 
dari semua kolom pada data yang bersifat numerik, terdiri dari mean, median, modus, standar deviasi, variansi, range, nilai minimum, maksimum, kuartil, IQR, skewness dan kurtosis. Boleh juga ditambahkan deskripsi lain.

In [75]:
# Import data
import pandas as pd
gandum = pd.read_csv("Gandum.csv")
gandum.columns = ['id','Daerah','SumbuUtama','SumbuKecil','Keunikan','AreaBulatan','Diameter','KadarAir','Keliling','Bulatan','Ransum','Kelas']
gandum.head()

id  Daerah  SumbuUtama  SumbuKecil  Keunikan  AreaBulatan   Diameter  \
0   2    4176  109.348294   49.837688  0.890098         4277  72.918093   
1   3    4555  114.427991   52.151207  0.890105         4706  76.155145   
2   4    4141  108.701191   49.457349  0.890499         4236  72.611879   
3   5    5273  122.747869   55.757848  0.890876         5431  81.937733   
4   6    5371  123.943530   56.235398  0.891145         5531  82.695644   

   KadarAir  Keliling   Bulatan    Ransum  Kelas  
0  0.596231   260.346  0.774227  2.194088      1  
1  0.776641   279.606  0.732159  2.194158      1  
2  0.633180   260.478  0.766960  2.197877      1  
3  0.669842   302.730  0.723031  2.201446      1  
4  0.734144   297.114  0.764573  2.204013      1

In [76]:
print("------------- Mean -------------")
print(gandum.mean(numeric_only=True))
print("\n\n")
print("------------- Median -------------")
print(gandum.median(numeric_only=True))
#asumsi NaN tidak bisa menjadi mode
print("\n\n")
print("------------- Mode -------------")
print(gandum.mode(numeric_only=True, dropna=True))
print("\n\n")
print("------------- Std. Deviasi -------------")
print(gandum.std(numeric_only=True))
print("\n\n")
print("------------- Varians -------------")
print(gandum.var(numeric_only=True))
print("\n\n")
print("------------- Range -------------")
gandum_Describe = gandum.describe()
for column in gandum_Describe:
    print(f"{column:12} ", gandum_Describe[column]["max"] - gandum_Describe[column]["min"])
print("\n\n")
print("------------- Minimum -------------")
print(gandum.min(numeric_only=True))
print("\n\n")
print("------------- Maksimum -------------")
print(gandum.max(numeric_only=True))
print("\n\n")
print("------------- IQR -------------")
gandum_Describe = gandum.describe()
for column in gandum_Describe:
    print(f"{column:12} ", gandum_Describe[column]["75%"] - gandum_Describe[column]["25%"])
print("\n\n")
print("------------- Skewness -------------")
print(gandum.skew(numeric_only=True))
print("\n\n")
print("------------- Kurtosis -------------")
print(gandum.kurtosis(numeric_only=True))
print("\n\n")

------------- Mean -------------
id              251.000000
Daerah         4799.282565
SumbuUtama      116.020635
SumbuKecil       53.705716
Keunikan          0.878741
AreaBulatan    4935.010020
Diameter         77.755080
KadarAir          0.648320
Keliling        281.409028
Bulatan           0.761813
Ransum            2.150829
Kelas             1.503006
dtype: float64



------------- Median -------------
id              251.000000
Daerah         4733.000000
SumbuUtama      115.396744
SumbuKecil       53.727453
Keunikan          0.889996
AreaBulatan    4855.000000
Diameter         77.628878
KadarAir          0.625731
Keliling        279.955000
Bulatan           0.761403
Ransum            2.193132
Kelas             2.000000
dtype: float64



------------- Mode -------------
      id  Daerah  SumbuUtama  SumbuKecil  Keunikan  AreaBulatan   Diameter  \
0      2  3992.0   74.133114   39.906517  0.719916       3802.0  71.293564   
1      3  4881.0   74.364021   41.436419  0.725553       49

In [77]:
print("------------- Kuartil -------------")
gandum.describe().loc[['25%','50%','75%']]

------------- Kuartil -------------


id  Daerah  SumbuUtama  SumbuKecil  Keunikan  AreaBulatan   Diameter  \
25%  126.5  4042.5  104.063393   51.192400  0.863673       4166.5  71.743089   
50%  251.0  4733.0  115.396744   53.727453  0.889996       4855.0  77.628878   
75%  375.5  5492.5  129.055961   56.309070  0.907613       5650.0  83.625761   

     KadarAir  Keliling   Bulatan    Ransum  Kelas  
25%  0.572602  255.8540  0.732088  1.983917    1.0  
50%  0.625731  279.9550  0.761403  2.193132    2.0  
75%  0.726830  305.7865  0.796387  2.382048    2.0

### 2. Membuat Visualisasi plot distribusi
dalam bentuk histogram dan boxplot untuk setiap kolom numerik. Berikan uraian penjelasan kondisi setiap kolom berdasarkan kedua plot tersebut.

### 3. Menentukan setiap kolom numerik berdistribusi normal atau tidak
Gunakan normality test yang dikaitkan dengan histogram plot.

### 4. Melakukan test hipotesis 1 sampel, dengan menuliskan 6 langkah testing
   menampilkan juga boxplotnya untuk kolom/bagian yang bersesuaian.<br>
   <ul>
    <li>Nilai rata-rata Daerah di atas 4700?</li>
    <li>Nilai Rata-rata Sumbu Utama tidak sama dengan 116?</li>
    <li>Nilai Rata-rata 20 baris pertama kolom Sumbu Kecil bukan 50?</li>
    <li>Proporsi nilai Diameter yang lebih dari 85, adalah tidak sama dengan 15% ?</li>
    <li>Proporsi nilai Keliling yang kurang dari 100, adalah kurang dari 5% ?</li>
   </ul>

### 5. Melakukan test hipotesis 2 sampel, dengan menuliskan 6 langkah testing 
menampilkan juga boxplotnya untuk kolom/bagian yang bersesuaian.
<ul>
    <li>Data kolom AreaBulatan dibagi 2 sama rata: bagian awal dan bagian akhir kolom. Benarkah rata-rata kedua bagian tersebut sama?</li>
    <li>Data kolom Kadar Air dibagi 2 sama rata: bagian awal dan bagian akhir kolom. Benarkah rata-rata bagian awal lebih besar dari pada bagian akhir sebesar 0.2?</li>
    <li>Rata-rata 20 baris pertama kolom Bulatan sama dengan 20 baris terakhirnya?</li>
    <li>Proporsi nilai bagian awal Ransum yang lebih dari 2, adalah lebih besar daripada, proporsi nilai yang sama di bagian akhir Ransum?</li>
    <li>Bagian awal kolom Diameter memiliki variansi yang sama dengan bagian akhirnya?</li>
</ul>

### 6. Test korelasi
tentukan apakah setiap kolom non-target berkorelasi dengan kolom target, dengan menggambarkan juga scatter plot nya. Gunakan correlation test.